internal/hcsoci/cmd.go

```diff
@@ -0,0 +1,318 @@
+package hcsoci
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/cow"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sys/windows"
+)
+
+// Cmd represents a command being prepared or run in a process host.
+type Cmd struct {
+	// Host is the process host in which to launch the process.
+	Host cow.ProcessHost
+
+	// OS is the operating system of the process host. FUTURE: Move to cow.ProcessHost.
+	OS string
+
+	// InUVM is true if Host is a utility VM. FUTURE: Move to cow.ProcessHost.
+	InUVM bool
```

Comment:
Same here. Why isn't this based on whether my ProcessHost is a UVM or a Container? IE: `Host` is the target of the command.

Reply:
We would need to add this to ProcessHost. We can do that, but it will be a bit of a pain to add to hcs.System because we will have to query the HCS. I'd like to leave that for another day.

Reply:
OK never mind I think I got it working.